packages/ag-charts-community/src/chart/agChartV2.ts

```diff
@@ -396,8 +397,12 @@ function applyChartOptions(chart: Chart, processedOptions: ProcessedOptions, use
 
     const seriesOpts = processedOptions.series as any[];
     const seriesDataUpdate = !!processedOptions.data || seriesOpts?.some((s) => s.data != null);
-    const otherRefreshUpdate = processedOptions.legend ?? processedOptions.title ?? processedOptions.subtitle;
-    forceNodeDataRefresh = forceNodeDataRefresh || seriesDataUpdate || !!otherRefreshUpdate;
+    const legendKeys = getLegendKeys();
+    const optionsHaveLegend = Object.values(legendKeys).some(
+        (legendKey) => (processedOptions as any)[legendKey] != undefined
+    );
+    const otherRefreshUpdate = Boolean(processedOptions.title ?? processedOptions.subtitle);
```

Comment:
Nit: Can we just use a straightforward expression rather than relying on Truthy evaluation? e.g.:
```
const otherRefreshUpdate = processedOptions.title != null || processedOptions.subtitle != null;
```